### PR TITLE
Add cmd/gopherlings exercise runner

### PR DESCRIPTION
**Pull request by jdkaplan**
Thanks for starting this project! I had a lot of fun with Rustlings, and I'm excited to see the same kind of beginner content come to Go.

The `rustlings` command was a big part of what made that fun for me, so I wrote a small watcher for myself as I worked through Gopherlings. I saw that this helper program is in the roadmap for this project, so I tried to clean it up for general use.

I borrowed the Rustlings "I AM NOT DONE" comment (now with a Go pun!) for similar reasons:

1. It encourages learners to keep playing with finished exercises.
2. Identifying unfinished exercises by running them isn't always possible. For example, `024-method` already runs without errors, and `010-forever` is an infinite loop.

Go compiler output is generally terse, so I tried to separate `gopherlings` output from `go` output during the `watch` loop with spacing and colors. I saw some reports online that the usual screen-clearing escape sequence works on modern Windows, but I can't easily test it myself. I'd probably use [github.com/inancgumus/screen](https://github.com/inancgumus/screen) otherwise.

There's a bunch of future work available here:

- A `hint` subcommand or interactive hints during the `watch` command.
- Allow `run` to take an argument that runs a specific exercise.
- Add a subcommand to run all exercises and check them against expected output.
- Catch errors that prevent `gopherlings` from running exercises (like an incomplete Go installation).

Last thing: I wrote a lot of text like help strings and the big success message. Since they may not fit with the flavor or personality you want for this project, I'll gladly rewrite or delete any of it that you point out.

### CODE REVIEW

**Comment by soypat** [comment]
Hey there- 
First of all, big thanks for the help! I just tested your tool and it really is delightful to work through exercises with it. Overall I am pleased with it, colors, command help message+output and all. Great user experience feel to it.

I have but one qualm regarding running the exercises and that is that they currently run indefinitely (infinite loop for the `forever` problem) and can print out unlimited text, blowing up command line. 

1. This could be helped modifying the `Run` method to pass a `context.WithTimeout` to `execFile`, 5 seconds ought to do it I think, we don't have any `time` exercises yet. 
2. The latter could be fixed by `io.LimitReader` and Reader/Writer stream wrapping.

That said, these are not requirements to getting this PR merged. Far more important is adding instructions to the `README.md`. I'd remove the mention of `air` and replace it with installation instructions of `gopherlings` using something of the sort:

```shell
go install github.com/soypat/gopherlings/cmd/gopherlings@latest
```

So to recap: 
- Add gopherlings installation instructions to `README.md` and VERY basic usage example
- Add exec timeout, 5 seconds?
- Limit gopherling output to, say 10.000 bytes?

If you are not available for contributing these features let me know and I'll merge this gem and work on it sometime next week.

**Comment by jdkaplan** [comment]
> * Add gopherlings installation instructions to `README.md` and VERY basic usage example

Done! I also updated the roadmap while I was here.

> * Add exec timeout, 5 seconds?

Done! That was a great idea :smile: It made the error handling a bit more complicated but also a bit more precise.

> * Limit gopherling output to, say 10.000 bytes?

This is the one I didn't do. I think it's also a great idea, but I ran out of time to work on it myself today!

**Comment by soypat** [comment]
Looks AWESOME to me! Merging! Thanks @jdkaplan !
